public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -1339,7 +1340,23 @@ export class DashboardModel implements TimeModel {
     return this.originalDashboard;
   }
 
-  hasAngularPlugins(): boolean {
+  async hasAngularPlugins(): Promise<boolean> {
+    // Ensure all panel plugins are loaded before checking if they are Angular plugins
+    await Promise.all(
+      this.panels.map(async (panel) => {
+        if (panel.plugin) {
+          return;
+        }
+        try {
+          const plugin = await dispatch(loadPanelPlugin(panel.type));
+          await panel.pluginLoaded(plugin);
```

Comment:
this should be return, not await.

Reply:
Ah, makes sense. However, what about the `await dispatch` above it? Is it fine if we `await` it and `return` only the second promise?

Reply:
ah, missed that await, then it'll be blocking anyway, in this case you should

```
return const dispatch(loadPanelPlugin(panel.type)).then( plugin => panel.pluginLoaded(plugin))
```

to return a single promise.